src/Interpreters/executeQuery.cpp

```diff
@@ -564,6 +564,47 @@ void logQueryFinish(
         query_span->addAttributeIfNotZero("clickhouse.written_rows", elem.written_rows);
         query_span->addAttributeIfNotZero("clickhouse.written_bytes", elem.written_bytes);
         query_span->addAttributeIfNotZero("clickhouse.memory_usage", elem.memory_usage);
+
+        if (context)
+        {
+            std::string user_name = context->getUserName();
+            query_span->addAttribute("clickhouse.user", user_name);
+        }
+
+        if (settings[Setting::log_query_settings])
+        {
+            auto changed_settings_names = settings.getChangedNames();
+            for (const auto & name : changed_settings_names)
+            {
+                Field value = settings.get(name);
+                String value_str;
+
+                switch (value.getType())
```

Comment:
Let's use here `convertFieldToString` from `Common/FieldVisitorToString.h`

Reply:
updated the code to use convertFieldToString from Common/FieldVisitorToString.h